extensions/ql-vscode/src/view/results/raw-results-table.tsx

```diff
@@ -19,158 +18,157 @@ import { onNavigation } from "./results";
 import { tryGetResolvableLocation } from "../../common/bqrs-utils";
 import { ScrollIntoViewHelper } from "./scroll-into-view-helper";
 import { sendTelemetry } from "../common/telemetry";
+import { assertNever } from "../../common/helpers-pure";
 
-export type RawTableProps = ResultTableProps & {
+export type RawTableProps = {
+  databaseUri: string;
   resultSet: RawTableResultSet;
   sortState?: RawResultsSortState;
   offset: number;
 };
 
-interface RawTableState {
-  selectedItem?: { row: number; column: number };
+interface TableItem {
+  readonly row: number;
+  readonly column: number;
 }
 
-export class RawTable extends React.Component<RawTableProps, RawTableState> {
-  private scroller = new ScrollIntoViewHelper();
-
-  constructor(props: RawTableProps) {
-    super(props);
-    this.setSelection = this.setSelection.bind(this);
-    this.handleNavigationEvent = this.handleNavigationEvent.bind(this);
-    this.state = {};
-  }
-
-  private setSelection(row: number, column: number) {
-    this.setState((prev) => ({
-      ...prev,
-      selectedItem: { row, column },
-    }));
-    sendTelemetry("local-results-raw-results-table-selected");
-  }
-
-  render(): React.ReactNode {
-    const { resultSet, databaseUri } = this.props;
-
-    let dataRows = resultSet.rows;
-    if (dataRows.length === 0) {
-      return emptyQueryResultsMessage();
-    }
-
-    let numTruncatedResults = 0;
-    if (dataRows.length > RAW_RESULTS_LIMIT) {
-      numTruncatedResults = dataRows.length - RAW_RESULTS_LIMIT;
-      dataRows = dataRows.slice(0, RAW_RESULTS_LIMIT);
-    }
-
-    const tableRows = dataRows.map((row: ResultRow, rowIndex: number) => (
-      <RawTableRow
-        key={rowIndex}
-        rowIndex={rowIndex + this.props.offset}
-        row={row}
-        databaseUri={databaseUri}
-        selectedColumn={
-          this.state.selectedItem?.row === rowIndex
-            ? this.state.selectedItem?.column
-            : undefined
+export function RawTable({
+  databaseUri,
+  resultSet,
+  sortState,
+  offset,
+}: RawTableProps) {
+  const [selectedItem, setSelectedItem] = React.useState<
+    TableItem | undefined
+  >();
+
+  const scroller = React.useMemo(() => new ScrollIntoViewHelper(), []);
```

Comment:
[`useRef`](https://react.dev/reference/react/useRef) is probably better for this since it's made for keeping things such as this. The behavior is probably identical, but `useRef` will guarantee that this value is not cleared and will not cause a re-render.